com/williamfiset/algorithms/datastructures/linkedlist/DoublyLinkedList.java

```diff
@@ -77,6 +77,38 @@ public void addFirst(T elem) {
     size++;
   }
 
+  // Add an element at a specified index
+  public void addAt(int index, T data) throws Exception
+	{
+		if(index<0)
+		{
+			throw new Exception("Illegal Index");
+		}
+		if(index==0)
+		{
+			addFirst(data);
+			return;
+		}
+		
+		if(index==size)
+		{
+			addLast(data);
+			return;
+		}
+		
+		Node<T> temp=head;
+		for(int i=0;i<index-1;i++)
+		{
+			head=head.next;
```

Comment:
For consistency, I would keep the head pointer pointing at the head and let the temp pointer do the traversing. So here and below replace references to head with temp.

Reply:
I have done these changes. Thanks for the reviews.